evalml/automl/automl_search.py

```diff
@@ -686,7 +706,8 @@ def _add_result(self, trained_pipeline, parameters, training_time, cv_data, cv_s
             "high_variance_cv": high_variance_cv,
             "training_time": training_time,
             "cv_data": cv_data,
-            "percent_better_than_baseline": percent_better,
+            "percent_better_than_baseline_all_objectives": percent_better_than_baseline,
+            "percent_better_than_baseline": percent_better_than_baseline[self.objective.name],
```

Comment:
@freddyaboulton so in `cv_data`, do we have separate fields for the primary objective vs the additional ones? Let's just make sure we're consistent across our code. Relatedly, right now, does `percent_better_than_baseline_all_objectives` also include the primary objective?

I think this would be cleanest if `percent_better_than_baseline` were simply a dict here in `results`, and then in `full_rankings` when we make the rankings leaderboard, we extract that to make the column. But if that's not what we do in `cv_data`, let's punt on that. What do you think?

Reply:
@dsherry I just checked and `cv_data` has a field for the primary objective (`score`) and for all objectives (`all_objective_scores` - this includes the primary objective as well). I think that means that `percent_better_than_baseline` and `percent_better_than_baseline_all_objectives` follow the same pattern we use for scores. 

Maybe we keep it like this in this PR and I file an issue for consolidating scores and percent better into a single field in the results?

Reply:
That sounds good!